tests/test_filters.py

```diff
@@ -0,0 +1,73 @@
+from django.test import TestCase
+
+from sockpuppet.templatetags.sockpuppet import camelcase, pascalcase
+
+
+class CamelcaseFilterTest(TestCase):
+    def test_camelcase(self):
+        self.assertEqual(camelcase("Foo"), "foo")
+
+    def test_2_camelcase(self):
+        self.assertEqual(camelcase("foo_bar"), "fooBar")
+
+    def test_3_camelcase(self):
+        self.assertEqual(camelcase("foo_bar_baz"), "fooBarBaz")
+
+    def test_preserve_already_camelcase(self):
+        self.assertEqual(camelcase("foo"), "foo")
+
+    def test_preserve_already_3_camelcase(self):
+        self.assertEqual(camelcase("fooBarBaz"), "fooBarBaz")
+
+    def test_pascalcase2camelcase(self):
+        self.assertEqual(camelcase("FooBarBaz"), "fooBarBaz")
+
+    # def test_preserve_CAPITALS_camelcase(self):
+    #     self.assertEqual(camelcase("test_GUI_part"), "testGUIPart")
+
+    def test_correct_CAPITALS_camelcase(self):
+        self.assertEqual(camelcase("test_GUI_part"), "testGuiPart")
+
+    def test_2_mixed_camelcase(self):
+        self.assertEqual(camelcase("FooBarBaz_FooBarBaz"), "foobarbazFoobarbaz")
+
+    def test_2_reverse_camelcase(self):
+        self.assertEqual(camelcase("Foo_bar"), "fooBar")
+
+    def test_double_camelcase(self):
+        self.assertEqual(camelcase(camelcase("foo_bar")), "fooBar")
+
+
+class PascalcaseFilterTest(TestCase):
+    def test_pascalcase(self):
+        self.assertEqual(pascalcase("foo"), "Foo")
+
+    def test_2_pascalcase(self):
+        self.assertEqual(pascalcase("foo_bar"), "FooBar")
+
+    def test_3_pascalcase(self):
+        self.assertEqual(pascalcase("foo_bar_baz"), "FooBarBaz")
+
+    def test_preserve_already_pascalcase(self):
+        self.assertEqual(pascalcase("FooBar"), "FooBar")
+
+    def test_preserve_already_3_pascalcase(self):
+        self.assertEqual(pascalcase("FooBarBaz"), "FooBarBaz")
+
+    def test_camelcase2pascalcase(self):
+        self.assertEqual(pascalcase("fooBarBaz"), "FooBarBaz")
+
+    # def test_preserve_CAPITALS_pascalcase(self):
```

Comment:
Is there a reason for this being commented?

Reply:
ah - i just wanted you to know that I considered leaving the possibility of UPPERCASE blocks in the code - but this is such a small user case. Maybe it's better to just remove it, as the camelcase Js counterpart of stimulus-reflex is not able to do that.
I'll remove that.